src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -36,6 +38,41 @@ public boolean contains(Person toCheck) {
         return internalList.stream().anyMatch(toCheck::isSamePerson);
     }
 
+    /**
+     * Returns list of names similar to the given person's name.
+     */
+    public List<String> findNearDuplicates(Person toCheck) {
+        requireNonNull(toCheck);
+        String toCheckName = normalizeName(toCheck.getName().toString());
+
+        List<String> nearDuplicates = new ArrayList<>();
+        for (String name : getAllNames()) {
```

Comment:
I think this is a straightforward and clean way of checking for duplicates, but my concern is if the contact list is extremely long which may cause the for loop to take a while to iterate through all entries.

Reply:
I think it's fine. To do better would require us implementing a HashSet, and I think we can assume that our user's contact list won't be incredibly long (say, less than 100K names).